File: Week_07/G20190343020149/231.2-的幂.go
/*
 * @lc app=leetcode.cn id=231 lang=golang
 *
 * [231] 2的幂
 */

// @lc code=start
func isPowerOfTwo(n int) bool {
	return n > 0 && (n&(n-1)) == 0
}

// @lc code=end

File: Week_07/G20190343020145/231.2-的幂.java
//2的幂次方问题转化为，二进制数中仅有一个1，即清零操作只需一次
/*
class Solution {
    public boolean isPowerOfTwo(int n) {
        if(n <= 0) return false;
        if((n&(n - 1)) == 0)
            return true;
        return false;
        
    }
}
*/

class Solution {
    public boolean isPowerOfTwo(int n) {
        return n > 0 && (n &(n - 1)) == 0;
    }
}